### PR TITLE
Added support for special characters and white spaces in image name

### PR DESCRIPTION
**Pull request by ivanbarlog**
Since `$path` is retrieved from URL I just used `urldecode` so also images with whitespaces or special characters in filename will be rendered correctly.

### CODE REVIEW

**Comment by makasim** [comment]
Could you please add a comment, and a test?

**Comment by ivanbarlog** [comment]
Of course. What should be commented?

File: Controller/ImagineController.php [hunk 1]
```diff
@@ -76,6 +76,8 @@ public function __construct(
      */
     public function filterAction(Request $request, $path, $filter)
     {
+        $path = urldecode($path);
```
Comment (makasim) [inline Controller/ImagineController.php]: > Of course. What should be commented?

this line, the purpose of it

**Comment by ivanbarlog** [comment]
@makasim I've just had a look at your tests but I am sorry I am not sure where should I add my test and what should this test exactly do. Could you please help me with this? I mean find proper place for introducing new test.

I can understand that test should check whether some path with special characters or white spaces or both will be resolved. For example `obrázok č. 1.jpg`. But I am not sure that I follow tests in the suite.

**Comment by makasim** [comment]
https://github.com/liip/LiipImagineBundle/blob/master/Tests/Functional/Controller/ImagineControllerTest.php

**Comment by ivanbarlog** [comment]
@makasim Can you tell me what's wrong with Travis build? Thanks

File: Tests/Functional/Controller/ImagineControllerTest.php [hunk 2]
```diff
@@ -190,4 +190,24 @@ public function testShouldResolveWithCustomFiltersFromCache()
...
+
+    public function testShouldResolvePathWithSpecialCharactersAndWhiteSpaces()
+    {
+        $this->filesystem->dumpFile(
+            $this->cacheRoot.'/thumbnail_web_path/images/mačací obrázok.jpeg',
+            'anImageContent'
+        );
+
+        // we are calling url with encoded file name as it will be called by browser
+        $urlEncodedFileName = urlencode("mačací obrázok");
```
Comment (makasim) [inline Tests/Functional/Controller/ImagineControllerTest.php]: could you hard code url encoded version?

**Comment by ivanbarlog** [comment]
@makasim Hi, I didn't find pull-request merging strategy anywhere in this repo. Could you please provide some information about when this PR will be merged and released?

Thanks in advance

**Pull request merged**